crypto/sql_store.go

```diff
@@ -0,0 +1,507 @@
+// Copyright (c) 2020 Tulir Asokan
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package crypto
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"maunium.net/go/mautrix/crypto/olm"
+	"maunium.net/go/mautrix/id"
+)
+
+// WarnlnLogger is used to log warnings.
+type WarnlnLogger interface {
+	Warnln(...interface{})
```

Comment:
This should probably support the same logger as OlmMachine

Reply:
Sure, done.